backend/Services/CoordinateFetcher.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Internships\Services;
+
+use GuzzleHttp\Client;
+use Internships\Models\FetchAddress;
+
+class CoordinateFetcher
+{
+    protected string $mapboxToken;
+
+    public function __construct()
+    {
+        $this->mapboxToken = $_ENV["MAPBOX_TOKEN"];
+    }
+
+    public function coordinatesFromAddress(string $address = "", FetchAddress $addressObject = null): string
+    {
+        if ($addressObject) {
+            $address = $this->fetchAddressToString($addressObject);
+        }
+
+        $api = new Client();
```

Comment:
Why not in the constructor?

Reply:
The class is intended to be a form of manager that is instanced only once and could be extended. Maybe in future it will store all failed and successful attempts, and use them somewhere for statistics?

Either way, I don't see a reason to keep Client loaded all the time.

Alternatively I can turn this into simple object that will be constructed when needed and then discarded, because currently it doesn't have any functionality beyond fetching coordinates from single address.

Reply:
But why w can't use DI (dependency injection) if we are using it in the whole project?

Reply:
I think I'll use PHP-DI then for now. We can consider how to better handle this if the project ever will require multiple users.